### PR TITLE
Array shift

### PR DESCRIPTION
**Pull request by TheQuangNguyen**

### CODE REVIEW

**Comment by TheQuangNguyen** [comment]
- [X] Top-level README “Table of Contents” is updated
 - [X] Feature tasks for this challenge are completed
 - [X] Unit tests written and passing
     - X ] “Happy Path” - Expected outcome
     - [X] Expected failure
     - [X] Edge Case (if applicable/obvious)
 - [X] README for this challenge is complete
     - [X] Summary, Description, Approach & Efficiency, Solution
     - [X] Link to code
     - [X] Picture of whiteboard

File: code401challenges/src/main/java/code401challenges/ArrayShift.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package code401challenges;
+
+public class ArrayShift {
+
+    // Write a function called insertShiftArray which takes in an array and the value to be added.
```
Comment (jeffborda) [inline code401challenges/src/main/java/code401challenges/ArrayShift.java]: Nice use of comments here!

File: code401challenges/src/main/java/code401challenges/ArrayShift.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        return shiftArray;
+    }
+
+    // write a second function that removes an element from the middle index
+    // and shifts other elements in the array to fill the new gap.
+
+    public static int[] removeShiftArray(int[] arr) {
+
+        // checks if the input array is empty
```
Comment (jeffborda) [inline code401challenges/src/main/java/code401challenges/ArrayShift.java]: I like how you are thinking about what to do if you are given edge cases

File: code401challenges/src/test/java/code401challenges/ArrayShiftTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package code401challenges;
+
+import org.junit.Test;
+
+import static code401challenges.ArrayShift.*;
+import static org.junit.Assert.*;
+
+public class ArrayShiftTest {
+
```
Comment (jeffborda) [inline code401challenges/src/test/java/code401challenges/ArrayShiftTest.java]: Excellent choice of test cases

File: code401challenges/src/main/java/code401challenges/ArrayShift.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    // return an array with the new value added at the middle index.
+
+    public static int[] insertShiftArray(int[] arr, int numberToAdd) {
+        int[] shiftArray = new int[arr.length+1];
+        int middleIndex = shiftArray.length/2;
+        int pointer = 0;    // pointer to traverse through input array
+
+        // loop through the new array and insert values from the input array into new array
+        // unless the index is equals to middleIndex
+        for (int i = 0; i < shiftArray.length; i++) {
```
Comment (ncarignan) [inline code401challenges/src/main/java/code401challenges/ArrayShift.java]: why not use a j instead of an i
